### PR TITLE
Fix WS disconnection handling

### PR DESCRIPTION
**Pull request by ziopio**
The following reconnections scenarios are handled in this PR

- [x] The server closes the connection, gun_down is handled correctly
- [x] The gun process crashes, the monitor message is used to react to it
- [x] Ping timeout, if the server is not pinging, the client throws away the connections and re-attempts it.

This PR requires https://github.com/stritzinger/kraft/pull/3 to go in production

### CODE REVIEW

**Review by maehjam — APPROVED** [review]
Looks good. What would also be important to test, but maybe only manually is how `grisp_connect` handles a longer disconnect or a random sequence of different kinds of disconnects. Do we get some queue up of gun instances for example? I can't think of a test case, but manually we should just unplug and replug the ethernet cable a lot to test the handling of this kind of disconnect as well. (Is it the same as crashing gun? I don't think so.)

**Comment by maehjam** [comment]
Oh the previous comment only refers to the tests. I missed the commits before. Wait for another approval!

File: src/grisp_connect_log_server.erl
Comment (maehjam) [inline src/grisp_connect_log_server.erl:114]: `catch error:_ ` should work as well. We don't need to match for the stack trace.

File: src/grisp_connect_ws.erl
Comment (maehjam) [inline src/grisp_connect_ws.erl:82]: Add a log of level notice or warning here.

**Review by maehjam — CHANGES_REQUESTED** [review]
See inline comments, rest looks good.

**Comment by ziopio** [comment]
> Looks good. What would also be important to test, but maybe only manually is how `grisp_connect` handles a longer disconnect or a random sequence of different kinds of disconnects. Do we get some queue up of gun instances for example? I can't think of a test case, but manually we should just unplug and replug the ethernet cable a lot to test the handling of this kind of disconnect as well. (Is it the same as crashing gun? I don't think so.)

Quick network interrruptions are unhandled (undetected) by gun. So an API call will just timeout but gun will not generate errors. That is why the `ws_ping_timeout` I am introducing in this PR is so important. It forces gun to reconnect.

**Review by maehjam — APPROVED** [review]

**Pull request merged**
